### ISSUE TITLE
Conflict with Eclipse Code Formatter plugin

### ISSUE THREAD

**Issue by krasa** [bazelbuild/intellij#92]
See https://github.com/krasa/EclipseCodeFormatter/issues/140

It seems to me that you are overriding CodeStyleManager just for formatting your own new file type. That is a very bad solution - https://intellij-support.jetbrains.com/hc/en-us/community/posts/206117919/comments/206693509 

You should probably do this: http://www.jetbrains.org/intellij/sdk/docs/tutorials/custom_language_support/formatter.html

**Comment by brendandouglas** [comment]
Implementing FormattingModelBuilder is not an option when we're delegating to external tools.

We've explicitly tried to find a solution that is *composable* -- that is, allows multiple plugins to override the formatter for their own languages without affecting each other (provided they're applied to separate languages of course).

For example, we use this same delegate model to implement code style enforcement across several of our language plugins.

The problem with the solution in [ProjectCodeStyleInstaller](https://github.com/krasa/EclipseCodeFormatter/blob/aa9a73b1f8af508bf84279d6c012fefc49a7021f/src/java/krasa/formatter/plugin/ProjectCodeStyleInstaller.java) is that it doesn't compose -- only one plugin can implement such a scheme at a time, across all languages.

**Comment by krasa** [comment]
Right, FormattingModelBuilder is not very friendly for external formatters, I'll write to JetBrains to do something about it.  

The check in ProjectCodeStyleInstaller was for detecting conflicts with google-java-format plugin, I will just  reimplement the whole thing.

google-java-format will need some changes too, since this will not work if their plugin starts before any other -> https://github.com/google/google-java-format/blob/8e53c2fc23aa2635a67f969689d0b17b6042fa5e/idea_plugin/src/com/google/googlejavaformat/intellij/GoogleJavaFormatInstaller.java#L46
